src/drivers/model/src/lib.rs

```diff
@@ -20,4 +19,16 @@ pub trait Driver {
     fn pwrite(&mut self, data: &[u8], pos: usize) -> Result<usize>;
     /// Shutdown the device.
     fn shutdown(&mut self);
+    /// Reads the exact number of bytes to fill in the `data`.
+    /// Returns ok if `data` is empty.
+    fn pread_exact(&self, data: &mut [u8], pos: usize) -> Result<()> {
+        let mut r = 0;
+        while r < data.len() {
+            r += self.pread(&mut data[r..], pos + r)?;
```

Comment:
Do we want to guard against an infinite loop of reading 0 bytes?

Reply:
Added protection against reading 0 bytes.

Reply:
Thanks. Does zero bytes mean EOF? It doesn't mean time out or some other random failure condition (like C libraries like to alias together)? If our spec doesn't specify, one possibility is that we may wish to constrain the return type of `pread` to `Result<NonZeroUsize>` to force future implementers of `Driver` to give everything a well defined meaning. I might be over-thinking this, but this kind of ambiguity does have a habit of breeding bugs.